mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -0,0 +1,449 @@
+                .fetchSet((rec) -> new ClanMember(this.clanID,rec.value1(),manager));
+
+        this.members = Set.copyOf(bruh);
+
+        return bruh;
+    }
+
+    /**
+     * Adds a member to the clan. Note that this does not have a built in limit, API implementer needs to implement
+     * limit if we desire a limit on members
+     *
+     * Note to API implementer: This handles nothing regarding invites. If you want an invite system, it needs to come
+     * before this method.
+     *
+     * @param transaction The tx
+     * @param player player to add
+     * @return true if it went ok, false if the member is already in this or another clan or if the member is the leader
+     */
+    public boolean addClanMember(Transaction transaction, SolarPlayer player) {
+        ClanDataObject receiver = player.getData(ClansKey.INSTANCE);
+
+        if (receiver.isSimilar(leader)) {
+            return false;
+        }
+
+        int sec = transaction.getProperty(DSLContext.class)
+                .insertInto(CLANS_CLAN_MEMBERSHIP)
+                .columns(CLANS_CLAN_MEMBERSHIP.CLAN_ID,CLANS_CLAN_MEMBERSHIP.USER_ID)
+                .values(this.clanID,receiver.getUserId())
+                .onDuplicateKeyIgnore()
+                .execute();
+
+        if (sec != 1) {
+            return false;
+        } else {
+            receiver.setCachedClan(this);
+            return true;
+        }
+    }
+
+    /**
+     * Removes a member from the clan.
+     * @param transaction The tx
+     * @param player player to remove from the clan
+     * @return return true if it went ok, false if member is not part of clan or already in a clan.
+     * @throws IllegalArgumentException if member tried to remove was the leader of the clan
+     */
+    public boolean removeClanMember(Transaction transaction, SolarPlayer player) {
+        ClanDataObject receiver = player.getData(ClansKey.INSTANCE);
+        if (receiver.isSimilar(leader))  {
+            throw new IllegalArgumentException("Tried to remove leader of clan!");
+        }
+
+        int res = transaction.getProperty(DSLContext.class)
+                .deleteFrom(CLANS_CLAN_MEMBERSHIP)
+                .where(CLANS_CLAN_MEMBERSHIP.CLAN_ID.eq(this.clanID).and(CLANS_CLAN_MEMBERSHIP.USER_ID.eq(receiver.getUserId())))
+                .execute();
+
+        if (res != 1)  {
+            return false;
+        } else {
+            receiver.setCachedClan(null);
+            return true;
+        }
+    }
+
+    /**
+     * Adds clan as an ally
+     *
+     * @param transaction the tx
+     * @param receiver the receiver
+     * @throws IllegalArgumentException if receiver is already allied with this clan or another clan, or if sender already has an ally.
+     * @return true if ok, false if clans are already allied or already have separate allies in the table.
+     */
+    public boolean addClanAsAlly(Transaction transaction, Clan receiver) {
```

Comment:
You need to review this method specification and its implementation. There are a few things to fix with it.